File: DataStructure/1. Two Sum.py
# Given an array of integers nums and an integer target, return indices of the two numbers such that they add up to target.

# You may assume that each input would have exactly one solution, and you may not use the same element twice.

# You can return the answer in any order.

nums = [2,7,11,15]

target = 9

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        temp = {}
        for index, num in enumerate(nums):
            temp[num] = index
        for index, num in enumerate(nums):
            if target - num in temp and index != temp[target - num]:
                return [index, temp[target - num]]

class Solution:
    def twoSum(self, nums: List[int], target: int) -> List[int]:
        for i in range(len(nums)):
            for j in range(1 + i, len(nums)):
                if nums[i] + nums[j] == target:
                    return [i, j]

File: LeetCode/DataStructure/1. Two Sum.cs
int[] nums = new int[] {2,7,11,15};

int target = 9;
public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        int[] array = new int[2];
        for(int i = 0; i < nums.Length; i++)
        {
            for(int j = 1 + i; j < nums.Length; j++)
            {
                if (nums[i] + nums[j] == target)
                {
                    array[0] = i;
                    array[1] = j;        
                } 
            }
        }
        return array; 
    }
}

public class Solution {
    public int[] TwoSum(int[] nums, int target) {
        var numsDict = new Dictionary<int, int>();
        var result = new List<int>();
        
        for (var i = 0; i < nums.Length; i++)
        {
            numsDict.Add(i, nums[i]);
        }
        
        for (var i = 0; i < nums.Length; i++)
        {
            if (numsDict.ContainsValue(target - nums[i]) && i != Array.IndexOf(nums, target - nums[i]))
            {
                result.Add(i);
                result.Add(Array.IndexOf(nums, target - nums[i]));
                break;
            }
        }
        return result.ToArray();
    }
}
